### PR TITLE
Fixes passcode-flow regressions

### PR DESCRIPTION
**Pull request by trooper2013**
@brandonpage Please help test various scenarios. Thanks.

### CODE REVIEW

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m [hunk 1]
```diff
@@ -215,23 +215,21 @@ + (void)validateTimer
...
-        } else {
-            // '0' is a special case.  We can't turn off passcodes unless all other users' passcode policies
-            // are also off.
-            if (![SFSecurityLockout nonCurrentUsersHavePasscodePolicy]) {
-                [SFSecurityLockout clearAllPasscodeState];
-                [SFSecurityLockout unlock:YES action:SFSecurityLockoutActionPasscodeRemoved];
-                return;
-            }
-        }
++ (BOOL)doesNotNeedPasscodeFlow:(NSUInteger)newLockoutTime {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m]: Can we switch this to `needsPasscodeFlow` instead? The double negation is kinda confusing :-)

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m [hunk 2]
```diff
@@ -215,23 +215,21 @@ + (void)validateTimer
...
+    return (newLockoutTime == 0)  && ![self usersHavePasscodePolicy];
+}
+
++ (BOOL)hasNewPasscodePolicy:(NSUInteger)newLockoutTime passcodeLength:(NSUInteger)newPasscodeLength {
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m]: Maybe we should call this something like `needsPasscodePolicyChange` instead. Anything entering this method already means there's a new login with a new passcode policy. We're only evaluating if the master passcode policy needs to change to reflect this new policy.

File: libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m [hunk 3]
```diff
@@ -215,23 +215,21 @@ + (void)validateTimer
...
-            // '0' is a special case.  We can't turn off passcodes unless all other users' passcode policies
-            // are also off.
-            if (![SFSecurityLockout nonCurrentUsersHavePasscodePolicy]) {
-                [SFSecurityLockout clearAllPasscodeState];
-                [SFSecurityLockout unlock:YES action:SFSecurityLockoutActionPasscodeRemoved];
-                return;
-            }
-        }
++ (BOOL)needsPasscodeFlow:(NSUInteger)newLockoutTime {
+    return newLockoutTime !=0 || [self usersHavePasscodePolicy];
```
Comment (bhariharan) [inline libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m]: Should this be `newLockoutTime` or `securityLockoutTime`?

Comment (trooper2013) [reply libs/SalesforceSDKCore/SalesforceSDKCore/Classes/Security/SFSecurityLockout.m]: Should be the newValue received from a login flow.

**Pull request merged**